libtransmission/torrent-files.cc

```diff
@@ -361,15 +366,17 @@ namespace
 {
     switch (ch)
     {
+    case ':':
+#if !defined(__APPLE__) // only ':' is reserved on APFS
```

Comment:
What about HFS?

Reply:
Same. While not an official source, we have:
https://www.reddit.com/r/MacOS/comments/rvilz7/reserved_file_names/
>The maximum filename length for HFS+ and APFS is 255 UTF-8 characters, and the reserved characters of HFS+ and APFS are /, :, CR (Carriage Return) and LF (Line Feed).

I've addressed CR and LF in the `> 31` catch-all for all control characters.